### PR TITLE
Have a way to show what version we are running

### PR DESCRIPTION
**Pull request by limed**

### CODE REVIEW

**Comment by limed** [comment]
Fixes issue #37

File: version.go [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package main
```
Comment (gozer) [inline version.go]: generated file doesn't belong in source-control

File: version.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+set -e
+
+cd "$(dirname "$0")"
+RELEASE=0
+VERSION="1.0"
+if REV=$(git rev-parse --short HEAD); then
```
Comment (gozer) [inline version.sh]: Using *git describe --tags* would be prettier I believe, i.e.
```
v1.2.2-125-gad17b8c
```

**Comment by gozer** [comment]
Generally, wouldn't it be simpler to get away without the go generate step and instead just use the top-level makefile to invoke something like:

```diff
diff --git a/Makefile b/Makefile
index a55ce28..d79071e 100644
--- a/Makefile
+++ b/Makefile
@@ -3,10 +3,13 @@
 NAME=nubis-user-management
 GO15VENDOREXPERIMENT := 1
 
+VERSION=$(shell git describe --always --tags --dirty)
+FLAGS=-X main.Version=$(VERSION)
+
 build:
        rm -rf build && mkdir -p build
-       mkdir -p build/linux && GOOS=linux go build -o build/linux/$(NAME) ./*.go
-       mkdir -p build/darwin && GOOS=darwin go build -o build/darwin/$(NAME) ./*.go
+       mkdir -p build/linux && GOOS=linux go build -ldflags="$(FLAGS)" -o build/linux/$(NAME) ./*.go
+       mkdir -p build/darwin && GOOS=darwin go build -ldflags="$(FLAGS)" -o build/darwin/$(NAME) ./*.go
 
 fmt:
        gofmt -w=true $$(find . -type f -name '*.go')
```

**Comment by limed** [comment]
> Generally, wouldn't it be simpler to get away without the go generate step and instead just use the top-level makefile to invoke something like:

Would this allow me to do `./nubis-bastionsshkey -version` and it will print out the version if I do this?

File: version.sh (hunk 2 above)
Comment (limed) [inline version.sh]: What happens in the scenario that I don't have a tag?

File: version.go (hunk 1 above)
Comment (limed) [inline version.go]: I can manage this file statically if thats the preferred method which means I will just remove the version.sh script. All this does is makes it easy for me to update this particular file by using `go generate`

Regardless of the situation I will still need this file even if its generated or not

**Comment by gozer** [comment]
Pretty much, yup.

**Comment by limed** [comment]
Fixed up makefile to drive versioning through a makefile

**Pull request merged**